src/codeu/chat/util/Time.java

```diff
@@ -66,4 +66,10 @@ public String toString() {
 
   public static Time now() { return Time.fromMs(System.currentTimeMillis()); }
 
+  // Difference of end and start times.
+  // Ratio is the power of ten from seconds that the duration will be given in
+  // (e.i. ratio = -3 is milliseconds)
+  public static long duration(Time start, Time end, int ratio) {
+    return Math.round(Math.pow(10, -(3 + ratio)) * ((end.inMs() - start.inMs())));
```

Comment:
I guess it would be useful to keep the ratio parameter since there may be features we might have to add in the future that may be helpful to have time testing. If anything, we can remove it when it has no use to us by the end of the project. I don't understand what you mean by immutable class, though.